Makefile

```diff
@@ -55,118 +78,156 @@ else # "variables.mk" was included. Business as usual until the end of this file
+
 sim1: | build deps wakunode1
 	echo -e $(BUILD_MSG) "build/$@" && \
 		$(ENV_SCRIPT) nim sim1 $(NIM_PARAMS) waku.nims
 
-example1: | build deps
-	echo -e $(BUILD_MSG) "build/$@" && \
-		$(ENV_SCRIPT) nim example1 $(NIM_PARAMS) waku.nims
 
-test1: | build deps
-	echo -e $(BUILD_MSG) "build/$@" && \
-		$(ENV_SCRIPT) nim test1 $(NIM_PARAMS) waku.nims
+## Waku v2 targets
 
-docs: | build deps
+test2: | build deps deps2 installganache
 	echo -e $(BUILD_MSG) "build/$@" && \
-		$(ENV_SCRIPT) nim doc --accept --index:on --project --out:.gh-pages waku/waku.nim waku.nims
+		$(ENV_SCRIPT) nim test2 $(NIM_PARAMS) waku.nims
+	# the following command (pkill -f ganache-cli) attempts to kill ganache-cli process on macos  
+	# if we do not kill the process then it would hang there and causes issue in GitHub Actions macos job (the job never finsihes)
+	(([[ $(detected_OS) = macOS ]] && \
+		pkill -f ganache-cli) || true)
 
-# Waku v2 targets
-wakunode2: | build deps
+wakunode2: | build deps deps2
 	echo -e $(BUILD_MSG) "build/$@" && \
 		$(ENV_SCRIPT) nim wakunode2 $(NIM_PARAMS) waku.nims
 
-sim2: | build deps wakunode2
+example2: | build deps
+	echo -e $(BUILD_MSG) "build/$@" && \
+		$(ENV_SCRIPT) nim example2 $(NIM_PARAMS) waku.nims
+
+sim2: | build deps deps2 wakunode2
 	echo -e $(BUILD_MSG) "build/$@" && \
 		$(ENV_SCRIPT) nim sim2 $(NIM_PARAMS) waku.nims
 
-example2: | build deps
+scripts2: | build deps wakunode2
 	echo -e $(BUILD_MSG) "build/$@" && \
-		$(ENV_SCRIPT) nim example2 $(NIM_PARAMS) waku.nims
+		$(ENV_SCRIPT) nim scripts2 $(NIM_PARAMS) waku.nims
+
+wakubridge: | build deps deps2
+	echo -e $(BUILD_MSG) "build/$@" && \
+		$(ENV_SCRIPT) nim bridge $(NIM_PARAMS) waku.nims
+
+chat2: | build deps deps2
+	echo -e $(BUILD_MSG) "build/$@" && \
+		$(ENV_SCRIPT) nim chat2 $(NIM_PARAMS) waku.nims
+
+chat2bridge: | build deps deps2
+	echo -e $(BUILD_MSG) "build/$@" && \
+		$(ENV_SCRIPT) nim chat2bridge $(NIM_PARAMS) waku.nims
+
+## Waku v2 tooling targets
 
-# Waku tooling targets
 wakucanary: | build deps
 	echo -e $(BUILD_MSG) "build/$@" && \
 		$(ENV_SCRIPT) nim wakucanary $(NIM_PARAMS) waku.nims
 
 
+## Waku docs
+
+docs: | build deps
+	echo -e $(BUILD_MSG) "build/$@" && \
+		$(ENV_SCRIPT) nim doc --accept --index:on --project --out:.gh-pages waku/waku.nim waku.nims
+
+
+## RLN
+
+# control rln code compilation
+ifeq ($(RLN), true)
+NIM_PARAMS := $(NIM_PARAMS) -d:rlnzerokit
+else ifeq ($(CI), true)
+NIM_PARAMS := $(NIM_PARAMS) -d:rlnzerokit
+endif
+
+ifeq ($(RLNKILIC), true)
+NIM_PARAMS := $(NIM_PARAMS) -d:rln
+endif
+
+# control compilation of rln tests that require on chain interaction
+ifeq ($(ONCHAIN_RLN), true) 
+NIM_PARAMS := $(NIM_PARAMS) -d:onchain_rln
+else
+ifeq ($(CI), true) 
+ifeq ($(detected_OS), macOS)
+NIM_PARAMS := $(NIM_PARAMS) -d:onchain_rln
+endif
+endif
+endif
+
+
 installganache:
 ifeq ($(ONCHAIN_RLN), true) 
 	npm install ganache-cli; npx ganache-cli -p	8540	-g	0	-l	3000000000000&
```

Comment:
@staheri14 @s1fr0 `ganache-cli` node package is deprecated. Check the NPM page for more context: https://www.npmjs.com/package/ganache-cli

Additionally, we should move the "ganache-cli" logic in the Makefile to the test suite itself.

How we should proceed with this?